### DISCUSSION TITLE
cmp.ConfirmBehavior clarification

### DISCUSSION THREAD

**Discussion by smackesey** [hrsh7th/nvim-cmp#628] [Q&A] [answered]
I see that `cmp.confirm` takes a behavior argument of `cmp.ConfirmBehavior.Insert` or `cmp.ConfirmBehavior.Replace`. What exactly differentiates these behaviors? It's not documented in either `nvim-cmp.txt` or the `README`.

I thought `Replace` would replace the entire word under the cursor, like this:

```
# NOTE: cursor location indicated by "|"

bo|at  # invoke completion: { boat, bore, bong }
bo|re  # I would expect this if I select "bore" for completion with `Replace`
bore|at  # Instead I get this, which is what I would expect for `Insert`
```

**Comment by adlrwbr** [comment]
I agree with your expectations. Looking at the source code, it appears cmp.ConfirmBehavior.Insert is the default behavior, but I can't spot a difference between the two modes. I'm going to open an issue.

**Comment by hrsh7th** [comment, marked as answer]
See https://github.com/hrsh7th/nvim-cmp/issues/664
